src/olympia/compat/cron.py

```diff
@@ -25,102 +26,98 @@ def compatibility_report(index=None):
-                doc = docs[addon.id]
-                doc.update(id=addon.id, slug=addon.slug, guid=addon.guid,
-                           binary=addon.binary_components,
-                           name=unicode(addon.name), created=addon.created,
-                           current_version=current_version)
-                doc['count'] = chunk[addon.id]
-                doc.setdefault('top_95',
-                               defaultdict(lambda: defaultdict(dict)))
-                doc.setdefault('top_95_all', {})
-                doc.setdefault('usage', {})[app.id] = updates[addon.id]
-                doc.setdefault('works', {}).setdefault(app.id, {})
-
-                # Populate with default counts for all app versions.
-                for ver in versions:
-                    doc['works'][app.id][vint(ver['main'])] = {
-                        'success': 0,
-                        'failure': 0,
-                        'total': 0,
-                        'failure_ratio': 0.0,
-                    }
-
-                # Group reports by `major`.`minor` app version.
-                reports = (CompatReport.objects
-                           .filter(guid=addon.guid, app_guid=app.guid)
-                           .values_list('app_version', 'works_properly')
-                           .annotate(Count('id')))
-                for ver, works_properly, cnt in reports:
-                    ver = vint(floor_version(ver))
-                    major = [v['main'] for v in versions
-                             if vint(v['previous']) < ver <= vint(v['main'])]
-                    if major:
-                        w = doc['works'][app.id][vint(major[0])]
-                        # Tally number of success and failure reports.
-                        w['success' if works_properly else 'failure'] += cnt
-                        w['total'] += cnt
-                        # Calculate % of incompatibility reports.
-                        w['failure_ratio'] = w['failure'] / float(w['total'])
-
-                if app not in addon.compatible_apps:
-                    continue
-                compat = addon.compatible_apps[app]
-                d = {'min': compat.min.version_int,
-                     'max': compat.max.version_int}
-                doc.setdefault('support', {})[app.id] = d
-                doc.setdefault('max_version', {})[app.id] = compat.max.version
+    log.info(u'Generating Firefox compat report.')
+    latest = UpdateCount.objects.aggregate(d=Max('date'))['d']
+    qs = UpdateCount.objects.filter(addon__appsupport__app=amo.FIREFOX.id,
+                                    addon__disabled_by_user=False,
+                                    addon__status__in=amo.VALID_STATUSES,
+                                    addon___current_version__isnull=False,
+                                    date=latest)
+
+    updates = dict(qs.values_list('addon', 'count'))
+    for chunk in chunked(updates.items(), 50):
+        chunk = dict(chunk)
```

Comment:
Investigated. It's as I suspected, the first chunk is a poor man `.iterator()`. Because of transforms - and maybe cache-machine as well - `.iterator() is almost useless on our models, especially `Addon` which pulls so much data at once. The chunking makes the cron consume less memory by doing them 50 at a time.

The second chunk  is definitely weirder: it would be useful to not send everything to ES at once, but it's indexing individually instead of using bulk indexing, so the advantage is lost. I'll file another bug to get rid of it and use bulk indexing.